src/main/java/com/esri/samples/na/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -0,0 +1,230 @@
+
+          // create default service area task parameters
+          ListenableFuture<ServiceAreaParameters> serviceAreaTaskParametersFuture = serviceAreaTask.createDefaultParametersAsync();
+          serviceAreaTaskParametersFuture.addDoneListener(() -> {
+            try {
+              ServiceAreaParameters serviceAreaParameters = serviceAreaTaskParametersFuture.get();
+              // set the task parameters to have the task return polygons
+              serviceAreaParameters.setPolygonDetail(ServiceAreaPolygonDetail.HIGH);
+              serviceAreaParameters.setReturnPolygons(true);
+              // clear the default service area, and add service areas of 1 minute and 3 minutes travel time by car
+              serviceAreaParameters.getDefaultImpedanceCutoffs().clear();
+              serviceAreaParameters.getDefaultImpedanceCutoffs().addAll(Arrays.asList(1.0, 3.0));
+
+              // create query parameters used to select all facilities from the feature table
+              QueryParameters queryParameters = new QueryParameters();
+              queryParameters.setWhereClause("1=1");
+
+              // add all facilities to the service area parameters
+              serviceAreaParameters.setFacilities(facilitiesTable, queryParameters);
+
+              // find the service areas around the facilities using the parameters
+              ListenableFuture<ServiceAreaResult> serviceAreaResultFuture = serviceAreaTask.solveServiceAreaAsync(serviceAreaParameters);
+              serviceAreaResultFuture.addDoneListener(() -> {
+                try {
+                  // get the task results
+                  ServiceAreaResult serviceAreaResult = serviceAreaResultFuture.get();
+
+                  // display all the service areas that were found to the map view
+                  List<Graphic> serviceAreaGraphics = serviceAreasGraphicsOverlay.getGraphics();
+
+                  // iterate through all the facilities to get the service area polygons
+                  for (int i = 0; i < serviceAreaResult.getFacilities().size(); i++) {
+                    List<ServiceAreaPolygon> serviceAreaPolygonList = serviceAreaResult.getResultPolygons(i);
+                    // create a graphic for each available polygon, as there may be more than one for each service area
+                    for (int j = 0; j < serviceAreaPolygonList.size(); j++) {
+                      // create and show a graphics for the service area
+                      serviceAreaGraphics.add(new Graphic(serviceAreaPolygonList.get(j).getGeometry(), fillSymbols.get(j % fillSymbols.size())));
+                    }
+                  }
+
+                } catch (ExecutionException | InterruptedException e) {
+                  new Alert(Alert.AlertType.ERROR, "Error solving the service area task").show();
+                }
+                // hide the progress indicator after the task is complete
+                progressIndicator.setVisible(false);
```

Comment:
could be done in a finally block so that the indicator is removed whether you get an uncaught exception or not.